src/prefect/task_engine.py

```diff
@@ -392,6 +392,11 @@ def handle_success(self, result: R, transaction: Transaction) -> R:
         )
         if transaction.is_committed():
             terminal_state.name = "Cached"
+
+        if PREFECT_EXPERIMENTAL_ENABLE_CLIENT_SIDE_TASK_ORCHESTRATION:
+            if self.task_run.start_time and not self.task_run.end_time:
+                self.task_run.end_time = terminal_state.timestamp
```

Comment:
So I wasn't going to put this in, but it looks like the expected behavior if a task run say crashes during orchestration from `PENDING` -> `CRASHED`, we wouldn't want to assign an end time, because there is no start time/duration

Reply:
OHHHH you're smart